brave/brave-core
# Improve Android first run experience for better retention

Resolves https://github.com/brave/brave-browser/issues/41203
Resolves https://github.com/brave/brave-browser/issues/41205
Resolves https://github.com/brave/brave-browser/issues/41204

The implementation is hidden behind default-disabled Griffin flag `BraveDayZeroExperiment`, so can be A/B-tested.

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
If day zero feature is enabled,
- We are forcing `Set default browser` system dialog on first page of onboarding
- We are showing an onboarding page to `add search widget`

**Test 1**

1. Run brave with clean profile and apply adb shell "echo ' --enable-features=BraveDayZeroExperiment:variant/a' > /data/local/tmp/brave-test-day-zero-expt"
2. Open brave
3. Observe `Set default browser` system dialog shows up on first page of onboarding
4. Observe an onboarding page to `add search widget`

**Test 2**

1. Run brave with clean profile and apply adb shell "echo ' --enable-features=BraveDayZeroExperiment:variant/b' > /data/local/tmp/brave-test-day-zero-expt"
2. Observe `Set default browser` system dialog or `add search widget` onboarding should not be shown.
